com.unity.shadergraph/Editor/Drawing/Inspector/PropertyDrawers/ShaderInputPropertyDrawer.cs

```diff
@@ -212,7 +212,21 @@ void BuildPropertyFields(PropertySheet propertySheet)
             if(property == null)
                 return;
 
-            switch(property)
+            if (property.version < property.latestVersion && !ShaderGraphPreferences.allowDeprecatedBehaviors)
+            {
+                var typeString = property.propertyType.ToString();
+                HelpBoxRow help;
+                help = new HelpBoxRow(MessageType.Warning);
+                var label = new Label("DEPRECATED: Hover for info")
+                {
+                    tooltip = $"The {typeString} Property has new updates. This version maintains the old behavior. If you update a {typeString} Property, you can use Undo to change it back. See the {typeString} Property documentation for more information."
+                };
+                help.Add(label);
+                help.contentContainer.Add(new Button(() => property.ChangeVersion(property.latestVersion)) { text = "Update" });
```

Comment:
I don't like the super-long button; I'd rather it be on the same line or right-aligned with a normal width.

Reply:
Alex had made a comment to the contrary, but I have no preference for either

Reply:
Alex had suggested keeping the button on another row to make it visually distinct, but I can try and figure out the ui to make it not as awkwardly long

Reply:
Another row is OK, just right-aligned with a max-width set (and probably min-width too). 👍